### PR TITLE
`Development`: Increase client test coverage of metrics

### PR DESCRIPTION
**Pull request by coaxial1**
#### General
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).

### Motivation and Context

Increase coverage of a bunch of files in ``app/admin/metrics``.

### Test Coverage

| File | | Branch % | Lines % |
| - | - | - | - |
| metrics.component.ts | Before | 33 | 64 |
|  | After | 78 | 100 |
| jvm-threads.component.ts | Before | 20 | 24 |
| | After | 88 | 100
| metrics-system.component.ts | Before | 0 | 23 |
| | After | 100 | 100 |

#### Code Review
- [x] Review 1
- [x] Review 2

### CODE REVIEW

File: src/test/javascript/spec/component/admin/metrics/metrics-system.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            declarations: [MetricsSystemComponent],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(MetricsSystemComponent);
+                comp = fixture.componentInstance;
+            });
+    });
+
+    it('millisecond to duration conversion', () => {
```
Comment (b-fein) [inline src/test/javascript/spec/component/admin/metrics/metrics-system.component.spec.ts:22]:
```suggestion
    it('should convert milliseconds to durations', () => {
```

**Review by b-fein — APPROVED** [review]
Code looks good.

Nitpick: I left an inline comment where I think one test case could be renamed to match the general naming scheme.

File: src/test/javascript/spec/component/admin/metrics/metrics-system.component.spec.ts (hunk 1 above)
Comment (coaxial1) [reply src/test/javascript/spec/component/admin/metrics/metrics-system.component.spec.ts]: Added it 2ade43e :)

**Review by b-fein — APPROVED** [review]

**Review by akesfeden — DISMISSED** [review]
LGTM 👍

File: src/test/javascript/spec/component/admin/metrics/jvm-threads.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        fixture.detectChanges();
+
+        const button = fixture.debugElement.query(By.css('button.hand.btn.btn-primary.btn-sm'));
+        expect(button).not.toBe(null);
+
+        button.nativeElement.click();
+
+        expect(spy).toHaveBeenCalledTimes(1);
+        expect(spy).toHaveBeenCalledWith(MetricsModalThreadsComponent, { size: 'xl' });
+        expect(mockModalRef.componentInstance.threads).toBe(threads);
```
Comment (willergeorg) [inline src/test/javascript/spec/component/admin/metrics/jvm-threads.component.spec.ts:69]:
```suggestion
        expect(mockModalRef.componentInstance.threads).toEqual(threads);
```

File: src/test/javascript/spec/component/admin/metrics/metrics.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    it('should call refresh on init', () => {
+        const mockMetrics = {};
+        const mockThreadDump = { threads: [] };
+        jest.spyOn(service, 'getMetrics').mockReturnValue(of(mockMetrics as Metrics));
+        jest.spyOn(service, 'threadDump').mockReturnValue(of(mockThreadDump as ThreadDump));
+        expect(comp.updatingMetrics).toBe(true);
+        comp.ngOnInit();
+        expect(service.getMetrics).toHaveBeenCalled();
+        expect(service.threadDump).toHaveBeenCalled();
```
Comment (willergeorg) [inline src/test/javascript/spec/component/admin/metrics/metrics.component.spec.ts:35]: Would it be possible to specify the number of calls?

File: src/test/javascript/spec/component/admin/metrics/metrics.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        expect(comp.updatingMetrics).toBe(false);
+        expect(comp.metrics).toBe(mockMetrics);
+        expect(comp.threads).toBe(mockThreadDump.threads);
```
Comment (willergeorg) [inline src/test/javascript/spec/component/admin/metrics/metrics.component.spec.ts:38]:
```suggestion
        expect(comp.threads).toEqual(mockThreadDump.threads);
```

**Review by willergeorg — DISMISSED** [review]
Marked you some nits. But as the tests are currently running successfully, might not be necessary to incorporate them.

File: src/test/javascript/spec/component/admin/metrics/metrics.component.spec.ts (hunk 3 above)
Comment (coaxial1) [reply src/test/javascript/spec/component/admin/metrics/metrics.component.spec.ts]: Added in b28d473

File: src/test/javascript/spec/component/admin/metrics/jvm-threads.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            { threadState: ThreadState.TimedWaiting },
+            { threadState: ThreadState.Runnable },
+            { threadState: ThreadState.Waiting },
+            { threadState: ThreadState.Waiting },
+            { threadState: ThreadState.Waiting },
+        ] as Thread[];
+
+        comp.threads = threads;
+
+        expect(comp.threads).toBe(threads);
```
[the remark this answers is not in the document]
Comment (coaxial1) [reply src/test/javascript/spec/component/admin/metrics/jvm-threads.component.spec.ts]: Added in bcadd0f

File: src/test/javascript/spec/component/admin/metrics/metrics.component.spec.ts (hunk 4 above)
Comment (coaxial1) [reply src/test/javascript/spec/component/admin/metrics/metrics.component.spec.ts]: Added in bcadd0f

File: src/test/javascript/spec/component/admin/metrics/jvm-threads.component.spec.ts (hunk 2 above)
Comment (coaxial1) [reply src/test/javascript/spec/component/admin/metrics/jvm-threads.component.spec.ts]: Added in bcadd0f

**Review by willergeorg — APPROVED** [review]
Thanks for incorporating my suggestions. Code is fine for me now!

**Review by Hialus — APPROVED** [review]
Code looks good 👍🏻

**Pull request merged**